LunrCoreLmdb/LmdbIndex.cs

```diff
@@ -0,0 +1,329 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using LightningDB;
+using Lunr;
+
+namespace LunrCoreLmdb
+{
+    public sealed class LmdbIndex
+    {
+        public Lazy<LightningEnvironment> Env { get; }
+
+        public string Path { get; }
+
+        public LmdbIndex(string path)
+        {
+            Env = new Lazy<LightningEnvironment>(() =>
+            {
+                var config = new EnvironmentConfiguration
+                {
+                    MaxDatabases = DefaultMaxDatabases,
+                    MaxReaders = DefaultMaxReaders,
+                    MapSize = DefaultMapSize
+                };
+                var environment = new LightningEnvironment(path, config);
+                environment.Open();
+                CreateIfNotExists(environment);
+                return environment;
+            });
+            Path = Env.Value.Path;
+        }
+
+		#region Fields 
+
+        public bool AddField(string field, CancellationToken cancellationToken = default)
+        {
+			cancellationToken.ThrowIfCancellationRequested();
```

Comment:
You have some mixed tabs and spaces in this file. Please standardize on spaces.

Reply:
Fixed. But you might want to add a `.gitattributes` file to do this automatically?